silver_braintree/payment_processors.py

```diff
@@ -370,9 +373,59 @@ def execute_transaction(self, transaction):
 
         return self._charge_transaction(transaction)
 
+    def find_lost_transaction_id(self, transaction):
+        if transaction.data.get('braintree_id'):
+            return True
+
+        search_result = braintree.Transaction.search(
+            braintree.TransactionSearch.amount.is_equal(transaction.amount),
+            braintree.TransactionSearch.payment_method_token.is_equal(
+                transaction.payment_method.token
+            ),
+            braintree.TransactionSearch.created_at.between(
+                dateutil.parser.parse(transaction.data['requested_at']) - timedelta(seconds=1),
+                transaction.created_at + timedelta(seconds=61)
+            )
+        )
+
+        transactions = list(search_result.items)
+
+        def sync_transaction_id(unsynced_transaction, transaction_list):
```

Comment:
I thought using inner functions would be better than just having a long chain of lines of code, that don't have much reusability potential.
It's easier to track what the parent function returns IMO:
`return sync_transaction_id(transaction, narrow_transaction_results(transactions))`